arcgis-ios-sdk-samples/Analysis/Analyze hotspots/HotspotSettingsViewController.swift

```diff
@@ -28,14 +28,28 @@ class HotspotSettingsViewController: UITableViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        updateDatePickerLimits()
+    }
+    
+    /// Set the date picker limits to maintain a valid state.
+    private func updateDatePickerLimits() {
+        let oneDayInSeconds: TimeInterval = 24 * 60 * 60
+        // keep the start date less than the end date
+        startDatePicker.maximumDate = endDatePicker.date - oneDayInSeconds
```

Comment:
Presumably the operators just call into `Date.addingTimeInterval(_:)`. The documentation for that function states "This only adjusts an absolute value. If you wish to add calendrical concepts like hours, days, months then you must use a Calendar." Could you look into that?

Reply:
Huh, I didn't know about those APIs. Done.

Reply:
Neither did I!